### ISSUE TITLE
📄 Issue 1 — Documentation

### ISSUE THREAD

**Issue by Cristiano2132** [Cristiano2132/model_track#22]
# 📄 Issue 1 — Documentation & API Design: Correlation & Dependency Module

## Description

Define the theoretical and architectural foundation for the `stats.correlation` module, establishing both the statistical methodology and the public API contract for correlation and dependency metrics.

This issue formalizes the design decisions that will guide the implementation of continuous, categorical, and mixed-type dependency metrics.

___

## Goal

Provide:

1. A clear theoretical foundation for correlation and dependency metrics.
2. A formal API specification for the correlation module.
3. A standardized return contract using a structured DTO.
4. Explicit handling policies for edge cases and reproducibility.

This issue must produce an implementation-ready specification.

___

# 📚 Statistical Documentation

## Tasks

- [x] Explain correlation vs dependency
- [x] Document Pearson correlation (assumptions, linearity, normality)
- [x] Document Spearman rank correlation (monotonic relationships)
- [x] Document Cramér’s V (with bias correction — specify formula used)
- [x] Document Mutual Information (estimator choice and properties)
- [x] Provide assumptions, limitations, and interpretation guidance
- [x] Include academic references

## Acceptance Criteria

- Clear distinction between correlation and dependency
- Assumptions explicitly stated
- Bias correction formula for Cramér’s V specified
- Mutual Information estimator documented
- Usage recommendations provided
- References included (Agresti, Cover & Thomas, sklearn documentation)

___

# 🏗 API & Architecture Specification

This section defines the engineering contract of the module.

___

## 📦 Module Structure

stats/
└── correlation/
├── base.py
├── continuous.py
├── categorical.py
├── mixed.py
├── analyzer.py
└── init.py

___

## 🧠 Strategy Pattern

Each metric must implement:

```python
class CorrelationStrategy(ABC):
    def supports(self, x: pd.Series, y: pd.Series) -> bool:
        ...

    def compute(self, x: pd.Series, y: pd.Series) -> CorrelationResult:
        ...
```

The analyzer module orchestrates strategy selection.

___

📤 Return Contract (DTO)

All metrics must return a structured object:
```python
from dataclasses import dataclass

@dataclass(frozen=True)
class CorrelationResult:
    metric: str                 # e.g. "pearson", "spearman", "cramers_v"
    value: float                # metric value
    p_value: float | None       # statistical significance when applicable
    method: str                 # explicit method used
    n_obs: int                  # number of observations used
    variable_types: tuple[str, str]  # e.g. ("continuous", "categorical")
```
Design Decisions
	•	value must always be deterministic.
	•	p_value must be provided when statistically meaningful.
	•	Mutual Information may return p_value=None.
	•	The object must be immutable (frozen=True).
	•	n_obs ensures auditability in pipelines.
	•	variable_types improves traceability and debugging.

___

🔍 Metric Selection Policy

The system must support:

Automatic detection

`correlate(x, y)`

Automatically selects the appropriate strategy based on variable types.

Explicit method selection

`correlate(x, y, method="pearson")`

Raises an error if the method is incompatible with variable types.

___

🧹 Missing Values Policy
	•	Default behavior: drop rows where either variable is NaN.
	•	Behavior must be documented explicitly.
	•	Handling must be deterministic.

___

🎯 Determinism & Reproducibility
	•	Mutual Information must fix random_state if estimator is stochastic.
	•	All results must be reproducible given identical inputs.

___

⚖ Normalization Policy
	•	Pearson → range [-1, 1]
	•	Spearman → range [-1, 1]
	•	Cramér’s V → range [0, 1]
	•	Mutual Information → explicitly document whether raw or normalized

Normalization decisions must be documented clearly.

___

🚨 Edge Case Policy

Document explicit behavior for:
	•	Constant variables
	•	Single-category variables
	•	Empty inputs
	•	High-cardinality categorical variables
	•	Perfect multicollinearity
	•	Extremely small sample sizes

___

🧪 Future Extensibility

The design must allow adding new dependency metrics without modifying existing strategies (Open/Closed Principle).

Potential future extensions:
	•	Distance Correlation
	•	Maximal Information Coefficient (MIC)
	•	HSIC
	•	Partial Correlation

___

✅ Acceptance Criteria (Engineering)
	•	Public API signature fully documented
	•	DTO contract defined and justified
	•	Strategy pattern documented
	•	Edge-case behavior explicitly defined
	•	Reproducibility policy documented
	•	Implementation-ready specification approved

__

📌 Notes

Suggested references:
	•	Agresti — Categorical Data Analysis
	•	Cover & Thomas — Elements of Information Theory
	•	Scikit-learn documentation on Mutual Information
